src/transactions/test/PathPaymentTests.cpp

```diff
@@ -70,2166 +70,1815 @@ assetPathToString(const std::deque<Asset>& assets)
-    auto const minBalance5 =
-        app->getLedgerManager().getLastMinBalance(5) + 10 * txfee;
-
-    auto const paymentAmount = minBalance3;
-    auto const morePayment = paymentAmount / 2;
-    auto const trustLineLimit = INT64_MAX;
-
-    // sets up gateway account
-    auto const gatewayPayment = minBalance2 + morePayment;
-    auto gateway = root->create("gate", gatewayPayment);
-
-    // sets up gateway2 account
-    auto gateway2 = root->create("gate2", gatewayPayment);
-
-    auto idr = makeAsset(gateway, "IDR");
-    auto cur1 = makeAsset(gateway, "CUR1");
-    auto cur2 = makeAsset(gateway, "CUR2");
-    auto usd = makeAsset(gateway2, "USD");
-    auto cur3 = makeAsset(gateway2, "CUR3");
-    auto cur4 = makeAsset(gateway2, "CUR4");
-
-    SECTION("transact more than INT64_MAX in a path payment")
-    {
-        auto a1 = root->create("A1", minBalance4);
-        auto a2 = root->create("A2", minBalance4);
-
-        a1.changeTrust(idr, trustLineLimit);
-        a1.changeTrust(cur1, trustLineLimit);
-        a1.changeTrust(cur2, trustLineLimit);
-
-        a2.changeTrust(idr, trustLineLimit);
-        a2.changeTrust(cur1, trustLineLimit);
-        a2.changeTrust(cur2, trustLineLimit);
-
-        for_versions_from(10, *app, [&] {
-            SECTION("issue more than INT64_MAX")
-            {
-                // in this test, gateway will issue 2 * INT64_MAX of cur2
-                gateway.pay(a1, idr, trustLineLimit);
-                gateway.pay(a2, cur1, trustLineLimit);
-
-                // the offers below are in the order that they will be taken
-
-                // offer to issue cur2 for idr
-                gateway.manageOffer(0, cur2, idr, Price{1, 1}, INT64_MAX,
-                                    MANAGE_OFFER_CREATED);
-
-                // a2 is buying cur2. This is necessary so a1 can get rid of the
-                // first INT64_MAX cur2 during the path payment so it can cross
-                // the issuers second offer to get another INT64_MAX cur2
-                a2.manageOffer(0, cur1, cur2, Price{1, 1}, INT64_MAX,
-                               MANAGE_OFFER_CREATED);
-
-                // offer to issue cur2 for cur1
-                gateway.createPassiveOffer(cur2, cur1, Price{1, 1}, INT64_MAX,
-                                           MANAGE_OFFER_CREATED);
-
-                REQUIRE(a1.getTrustlineBalance(cur2) == 0);
-                REQUIRE(a2.getTrustlineBalance(cur2) == 0);
-
-                // IDR -> CUR2 -> CUR1 -> CUR2
-                a1.pay(a1, idr, INT64_MAX, cur2, INT64_MAX, {cur2, cur1});
-
-                REQUIRE(a1.getTrustlineBalance(cur2) == INT64_MAX);
-                REQUIRE(a2.getTrustlineBalance(cur2) == INT64_MAX);
-            }
+        if (emitEvents)
+        {
+            cfg.INVARIANT_CHECKS = {".*"};
+            cfg.EMIT_CLASSIC_EVENTS = true;
+            cfg.BACKFILL_STELLAR_ASSET_EVENTS = true;
+        }
```

Comment:
instead of passing the boolean and running the test twice, should we just enable it in this test (like you had in other tests), since the tested logic is purely additive. this will also reduce big code churn in this PR.

Reply:
I initially did this because I thought there was value in running them separately, but I don't think that's true. I reverted this change.